### PR TITLE
docs: Initial setup of the user documentation

### LINKED ISSUE #41: Initial setup of the documentation for users
discussion: #33 

- Create the app within this monorepo
- Use [SveltePress](https://sveltepress.site/)
- Deploy on ~~Vercel (for now)~~ 
    - Netlify OSS plan ?
    - Cloudflare OSS plan ?
    - Vercel OSS plan (Currently being revamped)

### PR DESCRIPTION
**Pull request by ciscorn**
Close #41

What I Did:
- Ran `pnpx sv create` to initialize a SvelteKit project.
- Installed `@sveltepress/vite` and `@sveltepress/theme-default`.
- Replaced the `sveltekit` plugin with the `sveltepress` plugin in `vite.config.js`.
- Added minimal contents, project configs, and set up the nav bar.

### CODE REVIEW

**Comment by ciscorn** [comment]
@xeho91 Deployed on Cloudflare Pages!
https://svelte-docgen.pages.dev/

File: apps/docs/package.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+		"eslint-plugin-svelte": "^2.46.1",
+		"globals": "^15.13.0",
+		"prettier": "^3.4.2",
+		"prettier-plugin-svelte": "^3.3.2",
+		"svelte": "^5.14.3",
+		"svelte-check": "^4.1.1",
+		"typescript": "^5.7.2",
+		"typescript-eslint": "^8.18.1",
+		"vite": "^6.0.3",
+		"vitest": "^2.1.8"
```
Comment (xeho91) [inline apps/docs/package.json:36]: Aren't these already at the root of the workspace? Plus the `@eslint-compat`.

Except for `svelte`, but you can use [`"catalog:"`](https://pnpm.io/catalogs) as version as well.

File: package.json [hunk 2]
```diff
@@ -48,8 +48,14 @@
 		"typedoc-plugin-coverage": "3.4.0",
 		"typedoc-plugin-mdn-links": "4.0.4",
 		"typescript": "catalog:",
-		"typescript-eslint": "^8.18.0",
-		"vite": "catalog:",
+		"typescript-eslint": "^8.18.1",
+		"vite": "^6.0.3",
```
Comment (xeho91) [inline package.json:52]: Is moving from `catalog:` related to the necessity to include vite `v6` in overrides?

File: apps/docs/.prettierignore
Comment (xeho91) [inline apps/docs/.prettierignore:1]: Can it be at the root `.prettierignore`? Or there's a good reason for not doing it?

File: apps/docs/.gitignore
Comment (xeho91) [inline apps/docs/.gitignore:1]: Would it be okay to merge those with workspace root `.gitignore`?

**Review by xeho91 — CHANGES_REQUESTED** [review]
Yay, we're getting there! I left a couple questions just to clarify a few things.

**Comment by xeho91** [comment]
I've [raised a question on the SveltePress discord](https://discord.com/channels/994160573333913691/1079957021949952051/1319151168945000458) to widen peer dependency of Vite on their side.

File: package.json (hunk 2 above)
Comment (ciscorn) [reply package.json:52]: Yes, it seems that pnpm update automatically rewrites it.

File: apps/docs/.prettierignore
Comment (ciscorn) [reply apps/docs/.prettierignore:1]: Prettier does not look at .prettierignore files in ancestor directories (unlike .gitignore) 😢 

It is required to run Prettier inside each package.

https://github.com/prettier/prettier/issues/4081

File: apps/docs/.gitignore
Comment (ciscorn) [reply apps/docs/.gitignore:1]: That works better. I’ve merged them.

File: apps/docs/package.json (hunk 1 above)
Comment (ciscorn) [reply apps/docs/package.json]: I completely forgot to clean it up. Thank you! I’ve removed it.

**Review by xeho91 — APPROVED** [review]

File: pnpm-workspace.yaml [hunk 3]
```diff
@@ -2,10 +2,11 @@
 ---
 packages:
   - "packages/*"
+  - "apps/*"
 
 # https://pnpm.io/catalogs
 catalog:
   "dts-buddy": "^0.5.0"
-  "svelte": "^5.3.0"
+  "svelte": "^5.14.4"
```
Comment (ciscorn) [inline pnpm-workspace.yaml:10]: Update this to prevent a mix of esrap 1.3 and 1.2 (though I’m not sure why this fixes it).

**Pull request merged**